transactions.go

```diff
@@ -361,6 +382,104 @@ func (lw *LibWallet) DecodeTransaction(txHash []byte) (string, error) {
 	return string(result), nil
 }
 
+func (lw *LibWallet) queryTransactions(query storm.Query) (string, error) {
+	var transactions []Transaction
+
+	err := query.Find(&transactions)
+	if err != nil {
+		return "", nil
+	}
+
+	jsonEncodedTransactions, err := json.Marshal(&transactions)
+	if err != nil {
+		return "", err
+	}
+
+	return string(jsonEncodedTransactions), nil
+}
+
+func (lw *LibWallet) countTransactions(query storm.Query) (int, error) {
+
+	count, err := query.Count(&Transaction{})
+	if err != nil {
+		return -1, err
+	}
+
+	return count, nil
+}
+
+func (lw *LibWallet) FilterTransactions(txFilter int32) (string, error) {
+	query := lw.parseTxFilter(txFilter)
+	return lw.queryTransactions(query)
+}
+
+func (lw *LibWallet) CountTransactions(txFilter int32) (int, error) {
+	query := lw.parseTxFilter(txFilter)
+	return lw.countTransactions(query)
+}
+
+func (lw *LibWallet) DetermineTxFilter(txHash []byte) (int32, error) {
+	tx, err := lw.GetTransactionRaw(txHash)
+	if err != nil {
+		return -1, nil
+	}
+	if tx.Type != TxTypeRegular {
+		return TxFilterStaking, nil
+	}
+
+	switch tx.Direction {
+	case TxDirectionSent:
+		return TxFilterSent, nil
+	case TxDirectionReceived:
+		return TxFilterReceived, nil
+	default:
+		return TxFilterTransferred, nil
+	}
+}
+
+// - Helper Functions
+
+func (lw *LibWallet) parseTxFilter(txFilter int32) storm.Query {
```

Comment:
- method doesn't actually "parse" some data, can be renamed to say `makeQueryForTxFilter` or `prepareTxQuery`
- Instead of declaring `var query storm.Query` (line 444 below), you can make use a named return variable in this method declaration.